httpie/client.py

```diff
@@ -254,13 +254,7 @@ def make_request_kwargs(
     data = args.data
     auto_json = data and not args.form
     if (args.json or auto_json) and isinstance(data, dict):
-        if data:
-            data = json.dumps(data)
-        else:
-            # We need to set data to an empty string to prevent requests
-            # from assigning an empty list to `response.request.data`.
-            data = ''
-
+        data = json.dumps(data) if data else ''
```

Comment:
The comment is important to keep. Could you revert that change?

Reply:
sure I will update it today :)

Reply:
I would prefer to not change the code: the old version is more clear. Do you mind not altering that part in tha PR?